R/stat-summary.r

```diff
@@ -43,7 +43,6 @@
 #' @export
 #' @examples
 #' d <- ggplot(mtcars, aes(cyl, mpg)) + geom_point()
-#' d + stat_summary(fun.data = "mean_cl_boot", colour = "red", size = 2)
```

Comment:
Removing this example as ```fun.data="mean_cl_boot" ``` is referencing to ```Hmisc```, which is dependent on ```ggplot2```... this may be the cause of error: 

```R
Error in is.list(x) : attempt to apply non-function
Calls: <Anonymous> ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
```

Reply:
I don't think that is the case. Otherwise we would have gotten errors in previous `check()` results. And we have to make sure the tests and examples work fine with ggplot2 loaded as we will need that in the future. So please revert.

Reply:
I tested by removing this particular example, in all condition my ```cran``` check is passing on local machine, but when this example is present, I get the error: 

```R
d <- ggplot(mtcars, aes(cyl, mpg)) + geom_point()
> d + stat_summary(fun.data = "mean_cl_boot", colour = "red", size = 2)
> 
> # You can supply individual functions to summarise the value at
> # each x:
> d + stat_summary(fun.y = "median", colour = "red", size = 2, geom = "point")
Error in stopifnot(is.list(x), is.list(val)) : 
  attempt to apply non-function
Calls: <Anonymous> ... ggplot_build -> <Anonymous> -> f -> <Anonymous> -> stopifnot
```
You can recheck this on your machine, I also looked into ```Hmisc``` package as well, it has ```mean_cl_boot```. that's why when I removed ```ggplot2``` from system, and ran this example, I got: 
```R
> d + stat_summary(fun.data = "mean_cl_boot", colour = "red", size = 2)
Warning message:
Computation failed in `stat_summary()`:
Hmisc package required for this function 
```

I don't know why my commit is not being reflected in ```Travis``` builds, this example is still present there, I just wanted to run the test (just for now by removing this example) so, I can get if this is the sole reason for error, can you please look at the ```Travis logs```? 

Surely, I will revert back, if travis keep on throwing the same error after removing this example.

Reply:
Sorry, stupid mistake, forgot to make changes in ```man/stat_summary.Rd``` , ```Travis CI``` is passing on ```cran``` branch, please check.

Reply:
@faizan-khan-iit Your suggestion?

Reply:
@tdhock I need a second opinion here. 

As far as I can see, we still need to deal with these issues:

1. Make sure `ggplot2Animint` and `ggplot2` work without errors while being loaded in the same session
2. We could skip this one test that we are failing as this doesn't seem too major with regards to animint code base
3. Make minor changes to documentation in `ggplot2Animint` detailing the purpose of the package and the changes made
4. Merge `animint2` and `ggplot2Animint` into a single package
 
What do you think about this @tdhock? I think we could upload to CRAN after making relevant documentation changes (3). As for (1), (2) and (4), we could work on them afterwards. Any thoughts/suggestions?

Reply:
@vivekktiwari  Could you add some details to the README and the package documentation reflecting the changes made till now?

Reply:
@faizan-khan-iit, will update README with the changes made, can you let me know other places that needs to be updated regarding documentation? 

And should we merge this PR now?

Reply:
I think minor updates to vignettes should be enough

Reply:
@faizan-khan-iit should I build vignettes and push them too, or just make changes in ```vignettes``` folder?

Reply:
No need to make changes to `.gitignore`. But please make sure the vignettes build without errors

Reply:
@faizan-khan-iit, yes, definitely, I ran using ```devtools::build_vignettes()``` and it didn't throw any error, I have pushed changes made in ```vignettes``` folder.